packages/@react-spectrum/button/src/Button.tsx

```diff
@@ -153,23 +158,26 @@ function Button<T extends ElementType = 'button'>(props: SpectrumButtonProps<T>,
             : children}
           {isPending && <ProgressCircle
             aria-hidden="true"
+            aria-label={isPendingAriaLiveLabel}
             isIndeterminate
             size="S"
             UNSAFE_className={classNames(styles, 'spectrum-Button-circleLoader')}
             UNSAFE_style={{visibility: isProgressVisible ? 'visible' : 'hidden'}}
             staticColor={staticColor} />
           }
-          {/* Adding the element here with the same labels as the button itself causes aria-live to pick up the change in Safari.
-           Safari with VO unfortunately doesn't announce changes to *all* of its labels specifically for button
-           https://a11ysupport.io/tests/tech__html__button-name-change#assertion-aria-aria-label_attribute-convey_name_change-html-button_element-vo_macos-safari
-           The aria-live does cause extra announcements in other browsers. */}
-          {isPending && hasAriaLabel &&
-            <div aria-hidden="true" id={auxLabelId} aria-label={buttonProps['aria-label']} aria-labelledby={buttonProps['aria-labelledby']?.replace(buttonId, auxLabelId)} />
-          }
-          {isPending && <div
-            id={spinnerId}
-            aria-label={stringFormatter.format('pending')}
-            aria-labelledby={`${hasAriaLabel ? '' : iconId} ${hasAriaLabel ? auxLabelId : textId} ${spinnerId}`} />
+          {isPending &&
+            <>
+              <div aria-live={isFocused ? ariaLive : 'off'}>
+                {isProgressVisible &&
+                  <div role="img" aria-labelledby={isPendingAriaLiveLabelledby} />
+                }
+              </div>
+              {/* Adding the element here with the same labels as the button itself causes aria-live to pick up the change in Safari.
+              Safari with VO unfortunately doesn't announce changes to *all* of its labels specifically for button
+              https://a11ysupport.io/tests/tech__html__button-name-change#assertion-aria-aria-label_attribute-convey_name_change-html-button_element-vo_macos-safari
+              The aria-live may cause extra announcements in other browsers. */}
+              <div id={spinnerId} role="img" aria-label={isPendingAriaLiveLabel} />
+            </>
```

Comment:
The `aria-live` region is added when `isPending` is `true`, but we don't add the child element until `isProgressVisible` is also `true`, because the delay between adding `aria-live` element and its child helps ensure that the screen reader picks up on the live region before announcing the addition.

The `spinnerId` element is outside the `aria-live` region, so that it is not considered another addition to the live region requiring announcement.  

Since we are adding `aria-label={isPendingAriaLiveLabel}` and `aria-labelledby={isPendingAriaLiveLabelledby}` to the parent button when `isPending` is `true`, we don't need to set `aria-hidden="true"` on this element.

Reply:
i fought with this a few times trying to get it to work, some browsers seemed fine if live plus its contents were added at the same time, others not so much. great idea moving the live region down, I hadn't thought to do that